models/location_helpers.go

```diff
@@ -184,5 +199,66 @@ func CreateBackupLocation(q *reform.Querier, params CreateBackupLocationParams)
 	}
 
 	return row, nil
+}
+
+func testS3Config(c *S3LocationConfig) error {
+	// TODO: extract region and bucket
+	const (
+		region = "eu-central-1"
+		bucket = "extracted_bucket"
+	)
 
+	s3Config := &aws.Config{
+		Credentials: credentials.NewStaticCredentials(c.AccessKey, c.SecretKey, ""),
+		Region:      aws.String(region),
+	}
+	newSession, err := session.NewSession(s3Config)
+	if err != nil {
+		return err
+	}
+
+	s3Client := s3.New(newSession)
+	if _, err := s3Client.ListObjects(&s3.ListObjectsInput{Bucket: aws.String(bucket)}); err != nil {
+		if s3err, ok := err.(awserr.Error); ok {
```

Comment:
This can be maybe issue. You can have permissin to list objects and not to write to them and vice-versa.

Reply:
yeah, I understand that. As of now, I've replaced this approach with just checking if bucket exists and I have no other approaches in my mind, but I'm open to other suggestions.